File: Projects/Project #1 - Layout Exercise Project/app/src/main/java/com/example/myapplication/MainActivity.java
package com.example.myapplication;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.widget.EditText;
import android.widget.TextView;

import java.util.Calendar;

public class MainActivity extends AppCompatActivity {

    EditText dateOfBirth;
    TextView textView;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        dateOfBirth = (EditText) findViewById(R.id.editTextNumber);
        textView =(TextView) findViewById(R.id.textViewAge);
    }

    public void buttonFindAge(View view) {
        int dob = Integer.parseInt(dateOfBirth.getText().toString());
        Calendar calendar = Calendar.getInstance();
        int year = calendar.get(Calendar.YEAR);
        int age = (year - dob);
        textView.setText("age is:" + age);
    }
}

File: Projects/Project #1 - Layout Exercise Project/app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import android.content.Intent
import android.os.Binder
import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.Button
import android.widget.FrameLayout
import android.widget.ImageView
import android.widget.TextView
import androidx.constraintlayout.widget.ConstraintLayout
import androidx.databinding.DataBindingUtil
import com.example.myapplication.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity(), View.OnClickListener {

    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)
        init()
    }
    private fun init(){
        binding.button.setOnClickListener (this)
    }
    override fun onClick(view: View?){
        when(view?.id){
            binding.button.id ->{
                //click listener
                startActivity(Intent(this@MainActivity, StartActivity::class.java))
            }
        }
    }
}